src/utils/changeElementVOVisibility.js

```diff
@@ -1,12 +1,44 @@
 /**
  * This source file is part of the Swift.org open source project
  *
- * Copyright (c) 2021 Apple Inc. and the Swift project authors
+ * Copyright (c) 2022 Apple Inc. and the Swift project authors
  * Licensed under Apache License v2.0 with Runtime Library Exception
  *
  * See https://swift.org/LICENSE.txt for license information
  * See https://swift.org/CONTRIBUTORS.txt for Swift project authors
 */
+import TabManager from 'docc-render/utils/TabManager';
+
+const PREFIX = 'data-original-';
+const ARIA = 'aria-hidden';
+const TABINDEX = 'tabindex';
+
+function setOriginalValue(element, prop) {
+  // make sure that prop isn't cached already
+  let originalValue = element.getAttribute(PREFIX + prop);
+  if (originalValue === undefined || originalValue === null || originalValue === '') {
+    originalValue = element.getAttribute(prop) || '';
+    element.setAttribute(PREFIX + prop, originalValue);
+  }
+}
+
+function retrieveOriginalValue(element, prop) {
+  // get the cached property
+  const originalValue = element.getAttribute(PREFIX + prop);
+
+  // remove the prefixed attribute
+  element.removeAttribute(PREFIX + prop);
+
+  if (typeof originalValue === 'string') {
```

Comment:
I wonder why the check if its a `string` 🤔

Reply:
I tested this. If we call `element.getAttribute` on an attribute that doesn't exist, the return value is `null` and the type is therefore an object. 
In our case, the `cacheOriginalAttribute` ensures that even where there's no original value, we still set that attribute, just with an empty string. So in our case, the type of `originalValue` here should always be a string. However, if we had a bug where we somehow didn't set the attribute correctly, we would want to check the type here.

Reply:
Thanks for the explanation, @hqhhuang , maybe then it's easier to do something like this?

````js

if (originalValue === null) return;

if (originalValue.length) {
  element.setAttribute(prop, originalValue);
} else {
  // otherwise remove the attribute entirely.
  element.removeAttribute(prop);
}
````

Reply:
Cant you just check for `if(originalValue)` ? Its always a `string` or `null`, so a simple `truthy` check should work. 

We could also probably store `OG_PREFIX + prop` as a var in the function, so we dont concat it 3 times.

Reply:
This is a good idea!
`if (!element.hasAttribute(OG_PREFIX + prop)) return;`